sycl/test-e2e/ESIMD/performance_tests/instruction_count.py

```diff
@@ -0,0 +1,27 @@
+import os
+import re
+import sys
+
+
+def main(directory, max_count, target_file=None):
+    total_count = 0
+    pattern = re.compile(r"//\.instCount (\d+)")
+
+    for root, dirs, files in os.walk(directory):
+        for file in files:
+            if file.endswith(".asm") and (target_file is None or file == target_file):
+                print("Checking file: ", file)
```

Comment:
Do we need to add any error handling in the file IO? Previously we were seeing python just return 1 which wasn't useful for debugbing, so maybe we can print like "failed to open directory" "failed to open file" etc in the respective error cases